packages/client/src/generation/utils/buildDirname.ts

```diff
@@ -34,12 +35,23 @@ function buildDirnameFind(defaultRelativeOutdir: string, runtimePath: string) {
 
   return `
 const { findSync } = require('${runtimePath}')
+const fs = require('fs')
 
-const dirname = findSync(process.cwd(), [
+// some frameworks or bundlers replace or totally remove __dirname
+const hasDirname = typeof __dirname !== 'undefined' && __dirname !== '/'
```

Comment:
Would __dirname on Windows always be different than `'/'` actuallu?

Reply:
Looking at the comment, I assume that frameworks and bundlers that replace `__dirname` with `/` might do so regardless of the OS. That said, I'm not sure if it's necessary to check for this. If this path is fake, it will be caught by the `fs.existsSync` check below, otherwise it might be a legit real path (it would be an edge case, but something that might happen in someone's Docker container).

Reply:
Yes, this is really to catch bundlers like webpack that may replace it with `/` or might not even exist like in `electron`. If it turns out that `/` was indeed an existing path, then `findSync` just below will find it for sure.